### PR TITLE
Deprecate VersionType.FORCE

### PR DESCRIPTION
**Pull request by dakrone**
In #20995 we are making `force` versioning a hard failure for newly
created indices, so in 5.0+ we need to deprecate it.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/action/delete/DeleteRequest.java [hunk 1]
```diff
@@ -178,6 +181,9 @@ public long version() {
     }
 
     public DeleteRequest versionType(VersionType versionType) {
+        if (versionType == VersionType.FORCE) {
```
Comment (bleskes) [inline core/src/main/java/org/elasticsearch/action/delete/DeleteRequest.java]: I wondering if we should do it on validate? feels more natural there, as we won't need to wonder if there if we got all the paths to set version types

Comment (dakrone) [inline core/src/main/java/org/elasticsearch/action/delete/DeleteRequest.java]: Sure, I can move it to validation, I'll do that

**Comment by dakrone** [comment]
@bleskes pushed a commit that moves the deprecation logging into the validation instead of `.versionType`

**Review by bleskes — APPROVED** [review]
LGTM

**Pull request merged**